grakn-graph/src/main/java/ai/grakn/graph/internal/AbstractGraknGraph.java

```diff
@@ -858,39 +858,62 @@ public void validVertex(Vertex vertex){
     }
 
     //------------------------------------------ Fixing Code for Postprocessing ----------------------------------------
+    @Override
+    public boolean duplicateCastingsExist(String index, Set<ConceptId> castingVertexIds){
+        CastingImpl mainCasting = (CastingImpl) getMainConcept(index);
+        return getDuplicates(mainCasting, castingVertexIds).size() > 0;
+    }
+
     /**
-     * Merges the provided duplicate castings.
-     *
-     * @param castingVertexIds The vertex Ids of the duplicate castings
-     * @return if castings were merged, a commit is required and the casting index exists
+     * Returns the duplicates of the given concept
+     * @param mainConcept primary concept - this one is returned by the index and not considered a duplicate
+     * @param conceptVertexIds Set of Ids containing potential duplicates of the main concept
+     * @return true if the given set of Ids contains concepts that are duplicates of the main concept
      */
-    @Override
-    public boolean fixDuplicateCastings(String index, Set<ConceptId> castingVertexIds){
-        Set<CastingImpl> duplicated = castingVertexIds.stream()
-                .map(id -> this.<CastingImpl>getConceptRawId(id.getValue()))
+    private Set<? extends ConceptImpl> getDuplicates(ConceptImpl mainConcept, Set<ConceptId> conceptVertexIds){
+        Set<ConceptImpl> duplicated = conceptVertexIds.stream()
+                .map(id -> this.<ConceptImpl>getConceptRawId(id.getValue()))
                 //filter non-null, will be null if previously deleted/merged
                 .filter(Objects::nonNull)
                 .collect(toSet());
 
+        duplicated.remove(mainConcept);
+
+        return duplicated;
+    }
+
+    /**
+     * Given an index, get the concept associated with it
+     * @param index retrieve the concept associated with this index
+     * @return Concept representing the vertex at the given index
+     */
+    private ConceptImpl getMainConcept(String index){
         //This is done to ensure we merge into the indexed casting. Needs to be cleaned up though
-        CastingImpl mainCasting = getConcept(Schema.ConceptProperty.INDEX, index);
-        duplicated.remove(mainCasting);
+        return getConcept(Schema.ConceptProperty.INDEX, index);
```

Comment:
I also sort of disagree on this one. It's used in four places and if we inline it, we have to include that explanatory comment (of what it means to be the "main" concept) in each one.